### PR TITLE
Hook up wasi exit code to wasmer cli

### PR DESCRIPTION
**Pull request by lachlansneff**
Test by running:

```
> target/release/wasmer run examples/exit.wat
> echo $?
```

should echo "7".

Fixes (at least the exit code part) of #361.

### CODE REVIEW

**Comment by lachlansneff** [comment]
bors try

File: src/bin/wasmer.rs [hunk 1]
```diff
@@ -295,60 +297,81 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
...
+            );
 
-    let mut instance = module
-        .instantiate(&import_object)
-        .map_err(|e| format!("Can't instantiate module: {:?}", e))?;
+            let instance = module
+                .instantiate(&import_object)
+                .map_err(|e| format!("Can't instantiate module: {:?}", e))?;
 
-    webassembly::run_instance(
```
Comment (bjfish) [inline src/bin/wasmer.rs]: Maybe this method could be deprecated if it will no longer be called.

Comment (lachlansneff) [reply src/bin/wasmer.rs]: Yeah, this is a good idea.

File: src/bin/wasmer.rs [hunk 2]
```diff
@@ -295,60 +297,81 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
...
-        &module,
-        &mut instance,
-        abi,
-        if let Some(cn) = &options.command_name {
-            cn
+            let start: Func<(), ()> = instance.func("_start").map_err(|e| format!("{:?}", e))?;
```
Comment (bjfish) [inline src/bin/wasmer.rs]: There doesn't seem to be agreement on `_start` function yet (see linked wasi issue):
https://github.com/wasmerio/wasmer/issues/362

This is a little out of scope of this PR but maybe we could only call `_start` if it exists.

File: src/bin/wasmer.rs [hunk 3]
```diff
@@ -295,60 +297,81 @@ fn execute_wasm(options: &Run) -> Result<(), String> {
...
+                .args
+                .iter()
+                .map(|arg| arg.as_str())
+                .map(|x| Value::I32(x.parse().unwrap()))
+                .collect();
+            instance
+                .dyn_func("main")
+                .map_err(|e| format!("{:?}", e))?
+                .call(&args)
+                .map_err(|e| format!("{:?}", e))?;
```
Comment (bjfish) [inline src/bin/wasmer.rs]: Also sort of unrelated to this PR, I hope this becomes a command:
https://github.com/wasmerio/wasmer/pull/281#issuecomment-475689836
`wasmer call file.wasm function_name arg1 arg2`

Comment (lachlansneff) [reply src/bin/wasmer.rs]: This is a good idea, we should make an issue for that.

File: src/bin/wasmer.rs (hunk 2 above)
Comment (lachlansneff) [reply src/bin/wasmer.rs]: Yeah, I wasn't sure what to do for that. Maybe we should keep as is and make a different pr for that?

Comment (bjfish) [reply src/bin/wasmer.rs]: Yeah, I think a separate PR.

**Pull request merged**